src/object/cli_obj.c

```diff
@@ -1419,6 +1419,9 @@ obj_iod_sgl_valid(unsigned int nr, daos_iod_t *iods, d_sg_list_t *sgls,
 	int	i;
 	int	rc;
 
+	if (iods == NULL)
+		return -DER_INVAL;
```

Comment:
I think that the patch is fine, but i don't grok how this is related to the test that does not pass a NULL iod as far as i can see. Could you please elaborate?

Reply:
Most our using test code do not pass a NULL, but this ticket is for code sanity test (not real running tests) to make sure the API is secure even if some malfunction call(hacker, stress test, etc) passing NULL:
DAOS-1449: "If I pass a null pointer for the IOD parameter it appears to segfault."

SO, this fix is to make sure even NULL oid is passed in, DAOS wouldn't crash.